src/hooks/useTxHistory.ts

```diff
@@ -11,24 +12,29 @@ const useTxHistory = (
   error?: string
   loading: boolean
 } => {
+  const [filter] = useTxFilter()
+
   const { safe, safeAddress, safeLoaded } = useSafeInfo()
   const { chainId } = safe
 
-  // If pageUrl is passed, load a new history page from the API
+  // If filter exists or pageUrl is passed, load a new history page from the API
   const [page, error, loading] = useAsync<TransactionListPage>(
     () => {
-      if (!pageUrl || !safeLoaded) return
-      return getTransactionHistory(chainId, safeAddress, pageUrl)
+      if (!safeLoaded || !(filter.type || pageUrl)) return
+
+      return filter.type
+        ? fetchFilteredTxHistory(chainId, safeAddress, filter, pageUrl)
+        : getTransactionHistory(chainId, safeAddress, pageUrl)
     },
-    [chainId, safeAddress, safeLoaded, pageUrl],
+    [chainId, safeAddress, safeLoaded, pageUrl, filter],
     false,
   )
 
   // The latest page of the history is always in the store
   const historyState = useAppSelector(selectTxHistory)
 
   // Return the new page or the stored page
-  return pageUrl
+  return filter.type || pageUrl
```

Comment:
Why are we checking the type instead of just using a null when filter is not there?

Reply:
You defined the type with fixed properties. Do you suggest returning the following type instead (`undefined`)?

```ts
export type TxFilter = {
  type: TxFilterType | null
  filter: IncomingTxFilter | MultisigTxFilter | ModuleTxFilter // CGW filter
} | undefined
```